### DISCUSSION TITLE
java sdk 2.3.2 error

### DISCUSSION THREAD

**Discussion by roseZJA** [milvus-io/milvus#28698] [Announcement]
After instantiating MilvusServiceClient in Java SDK version 2.3.2, any operation such as createDatabase or insert results in a StatusRuntimeException with an internal panic! error. Switching to SDK Maven version 2.2.9 allows normal operation again. Now, there's a desire to utilize MetricType COSINE from Java SDK version 2.3.2 and above, which is not supported in lower versions. How to resolve the exceptions encountered in 2.3.2?

**Comment by yhmo** [comment]
Which version of your milvus?
There is a versions compatible matrix in the Java SDK README: https://github.com/milvus-io/milvus-sdk-java. Follow the matrix to install correct sdk version.

| Milvus version | Java SDK version |
| :------------: |:----------------:|
|     2.0      |      2.0.4       |
|     2.1      |   2.1.0-beta4    |
|     2.2.0 ~ 2.2.8      |      2.2.0 ~ 2.2.5       |
|     >= 2.2.9      |      2.2.7 ~ 2.2.15       |
|     2.3.x      |      2.3.3      |
